src/client.ts

```diff
@@ -105,11 +105,34 @@ export class SkynetClient {
    * The Skynet Client which can be used to access Skynet.
    *
    * @class
-   * @param [portalUrl] The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use ""
+   * @param [initialPortalUrl] The initial portal URL to use to access Skynet, if specified. A request will be made to this URL to get the actual portal URL. To use the default portal while passing custom options, pass "".
    * @param [customOptions] Configuration for the client.
    */
-  constructor(portalUrl: string = defaultPortalUrl(), customOptions: CustomClientOptions = {}) {
-    this.portalUrl = portalUrl;
+  constructor(initialPortalUrl = "", customOptions: CustomClientOptions = {}) {
+    // Get the initial portal URL, if not provided.
+    if (initialPortalUrl == "") {
+      initialPortalUrl = defaultPortalUrl();
+    }
+    // Kick off async request for the portal URL.
+    this.portalUrl = this.executeRequest({
+      ...customOptions,
+      method: "head",
+      url: initialPortalUrl,
+      endpointPath: "/",
+    }).then((response) => {
+      /* istanbul ignore next */
+      if (typeof response.headers === "undefined") {
+        throw new Error(
```

Comment:
these are all uncaught exceptions I think

Reply:
Throwing in `.then` is a rejected promise.

Reply:
what do you mean, my point is that if this error gets thrown, it will register as 
`Uncaught (in promise)` in the console log of whatever code or script that's calling this constructor, right?

Reply:
You may be right, idk. I think `reject` should be used in this case.